File: R/Code/regression_template.R
dataset = read.csv("Position_Salaries.csv")


#Split training and test sets
install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.8)
training_set =subset(dataset, split==TRUE)
test_set = subset(dataset, split==FALSE)

#Feature Scaling
# training_set[,2:3] = scale(training_set[,2:3])
# test_set[,2:3]= scale(test_set[,2:3])

# regressor
#create regressor

#predict with regression
y_pred = predict(regressor, newdata = data.frame(Level = 6.5))


#visualize regression
library(ggplot2)

ggplot() + geom_point(aes(x= dataset$Level, y=dataset$Salary), color='red')+ geom_line(aes(x=dataset$Level, y=predict(regressor, newdata= dataset)), color='blue')
+ ggtitle("Regression")+ xlab("Level")+ ylab("Salary")

#visualize regression smooth curve

x_grid= seq(min(dataset$Level), max(dataset$Level),0.1)
ggplot() + geom_point(aes(x= dataset$Level, y=dataset$Salary), color='red')+ geom_line(aes(x=x_grid, y=predict(regressor, newdata= data.frame(Level= x_grid))), color='blue')
+ ggtitle("Regression")+ xlab("Level")+ ylab("Salary")

File: Python/Code/regression_template.py
# -*- coding: utf-8 -*-
"""
Created on Sun Sep 16 14:13:00 2018

@author: Admin
"""

# -*- coding: utf-8 -*-
"""
Spyder Editor

This is a temporary script file.
"""

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt



dataset = pd.read_csv("D:\\Udemy_ML\\Polynomial_linear_regression\\Polynomial_Regression\\Position_Salaries.csv")
X= dataset.iloc[:,1:-1].values
y= dataset.iloc[:,len(dataset.columns)-1].values

"""from sklearn.preprocessing import Imputer
imputer = Imputer(missing_values='NaN', strategy='mean', axis=0)
imputer = imputer.fit(X[:,1:3])
X[:,1:3]= imputer.transform(X[:,1:3])"""

"""from sklearn.preprocessing import LabelEncoder, OneHotEncoder
oneHotEncoder = OneHotEncoder(categorical_features=[3])
labelEncoder_X= LabelEncoder()
X[:,3] = labelEncoder_X.fit_transform(X[:,3])
X=oneHotEncoder.fit_transform(X).toarray()"""


#avoiding dummy var trap
"""X=X[:,1:]
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test= train_test_split(X, y, test_size=0.2, random_state=0)"""

#Feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

from sklearn.linear_model import LinearRegression

#non-linear regression
#create regressor



#predict - nonlinear reg
y_pred=regressor.predict(6.5)



#visualize reg
plt.scatter(X, y, color='red')
plt.plot(X, regressor.predict(X), color='blue')
plt.title('Linear Regression')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

#visualize reg higher resolution
X_grid= np.arange(min(X), max(X),0.1)
X_grid=X_grid.reshape(len(X_grid),1)
plt.scatter(X, y, color='red')
plt.plot(X_grid, regressor.predict(X_grid), color='blue')
plt.title('Linear Regression')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()












